presto-docs/src/main/sphinx/cache/library.rst

```diff
@@ -0,0 +1,66 @@
+=====================
+Alluxio Cache Library
+=====================
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+To optimize Presto query latency when reading remote data using Hive connector,
+one can leverage a built-in cache library in each Presto worker.
+This cache, based on Alluixo client-side cache library,
+utilizes the local storage (such as SSD) with configurable capacity and locations,
+and requires no extra service.
+To understand the internals of this local cache and the benchmark results of latency improvement,
+please refer to this `article <https://prestodb.io/blog/2020/06/16/alluxio-datacaching>`_.
+Note that this is a **read-cache**, **completely transparent** from users
+and **fully managed** by each individual Presto worker node.
+To provide Presto an independent, distributed cache service for reads/writes and
+full-featured cache policies, please refer to :doc:`/cache/alluxio`.
+
+
+Setup
+-----
+
+To turn on this Alluxio cache library is very easy:
+simply include the following configuration in ``etc/catalog/hive.properties``
+and restart Presto workers.
+
+.. code-block:: none
+
+    hive.node-selection-strategy=SOFT_AFFINITY
+    cache.enabled=true
+    cache.type=ALLUXIO
+    cache.alluxio.max-cache-size=500GB
+    cache.base-directory=/tmp/alluxio-cache
+
+In the above example,
+
+* ``hive.node-selection-strategy=SOFT_AFFINITY`` instructs Presto scheduler to take data affinity into consideration when secheduling tasks to workers that enables meaningful data caching effectiveness.
```

Comment:
We should clarify that without soft affinity local data caching would be disabled. We should also mention the default NO_PREFERENCE here.

Reply:
Let's also mention following parameter which has huge impact of the worker affinity and cache effectiveness
Here are some other configurations which can useful:
Coordinator configuration (useful to configure the definition of a busy worker):
    Set max pending splits per task: node-scheduler.max-pending-splits-per-task
    Set max splits per node: node-scheduler.max-splits-per-node

Reply:
done